lib/Doctrine/ODM/PHPCR/UnitOfWork.php

```diff
@@ -3968,4 +3974,43 @@ public function invokeGlobalEvent($eventName, EventArgs $event)
             $this->eventManager->dispatchEvent($eventName, $event);
         }
     }
+
+    /**
+     * If the parent node has child restrictions, ensure that the given
+     * class name is within them.
+     *
+     * @param NodeInterface $parentNode
+     * @param string $classFqn
+     */
+    private function validateChildRestrictions(NodeInterface $parentNode, $classFqn)
+    {
+        $parentClass = $this->documentClassMapper->getClassName($this->dm, $parentNode);
+
+        if (null === $parentClass) {
+            return;
+        }
+
+        $metadata = $this->dm->getClassMetadata($parentClass);
+        $childRestrictions = $metadata->getChildRestrictions();
+
+        if (null === $childRestrictions) {
+            return;
+        }
+
+        if (array() === $childRestrictions) {
+            throw new OutOfBoundsException(sprintf(
+                'Document "%s" cannot have children',
+                $parentClass
+            ));
+        }
+
+        if (!in_array($classFqn, $childRestrictions)) {
```

Comment:
i think you need to loop and instanceof to allow base classes or interfaces in the list.

i guess specifying a namespace instead of a class makes little sense? and we could not handle instanceof in that scenario.